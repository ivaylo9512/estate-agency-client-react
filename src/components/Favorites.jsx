import { useDispatch } from "react-redux"
import { useSelector } from "react-redux";
import { getFavoritesState, getFavorites } from "../app/slices/favoritesSlice";
import { useEffect } from "react";

const Favorites = () => {
    const dispatch = useDispatch();
    const { isLoading, favorites } = useSelector(getFavoritesState);
    
    useEffect(() => {
        dispatch(getFavorites());
    },[])

    return(
        <div>
            {favorites.map(favorite => 
                <div key={favorite.id}>
                    {favorite.id}
                </div>
            )}
        </div>
    )
}
export default Favorites