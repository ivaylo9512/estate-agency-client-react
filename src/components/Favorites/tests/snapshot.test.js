import Favorites from '../Favorites';
import * as Redux from 'react-redux';
import { shallow } from 'enzyme';

describe('Favorites snapshot tests', () => {
    let selectorSpy;

    beforeAll(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());

        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (isAuth) => {
        selectorSpy.mockReturnValue(isAuth);

        return shallow(
            <Favorites />
        )
    }

    it('should match snapshot', () => {
        const wrapper = createWrapper({ favorites: ['first favorite', 'second favorrite', 'third favorite'] });

        expect(wrapper).toMatchSnapshot();
    })
})