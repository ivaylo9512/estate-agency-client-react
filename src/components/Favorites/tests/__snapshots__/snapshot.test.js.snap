// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites snapshot tests should match snapshot 1`] = `
<div>
  <div />
  <div />
  <div />
</div>
`;
